### PR TITLE
Home14

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/homework14/Main.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+public class Main {
+    private void countWords() {
+        Scanner scanner = new Scanner(System.in);
+        Stream.of(scanner.nextLine().toLowerCase().split("[^\\p{L}\\p{Digit}_]+"))
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
+                .entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new))
```
Comment (Sofasmile) [inline src/main/java/homework14/Main.java]: https://mate-academy.github.io/style-guides/java/java.html#s4.4-column-limit

Comment (IgorPrihodko) [reply src/main/java/homework14/Main.java]: Made some fixes. Check plz, is it right or better?)

File: src/main/java/homework14/Main.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                .nextLine()
+                .toLowerCase()
+                .split("[^\\p{L}\\p{Digit}_]+"))
+                .collect(Collectors
+                        .groupingBy(Function.identity(), Collectors.counting()))
+                .entrySet()
+                .stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors
+                        .toMap(Map.Entry
```
Comment (Sofasmile) [inline src/main/java/homework14/Main.java]: better:
```suggestion
                        .toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new))
```
or move into new line after " , "

File: src/main/java/homework14/Main.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                                        ::getKey,
+                                Map.Entry
+                                        ::getValue,
+                                (e1, e2) -> e2,
+                                LinkedHashMap
+                                        ::new))
+                .entrySet()
+                .stream()
+                .sorted(Comparator
+                        .comparingLong((ToLongFunction<Map.Entry<String, Long>>) Map.Entry
```
Comment (Sofasmile) [inline src/main/java/homework14/Main.java]:
```suggestion
                        .comparingLong((ToLongFunction<Map.Entry<String, Long>>) Map.Entry::getValue)
```

File: src/main/java/homework14/Main.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                                ::getValue)
+                        .reversed()
+                        .thenComparing(Map.Entry
```
Comment (Sofasmile) [inline src/main/java/homework14/Main.java]:
```suggestion
                        .thenComparing(Map.Entry::getKey))
```

File: src/main/java/homework14/Main.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/homework14/Main.java]: thx, done!

File: src/main/java/homework14/Main.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/homework14/Main.java]: done

File: src/main/java/homework14/Main.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/homework14/Main.java]: done

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**